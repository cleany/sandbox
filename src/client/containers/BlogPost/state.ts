import { Input } from './types';

export interface State {
  input?: Input;
}

export const initialState: State = {
  input: '',
};

export default State;
