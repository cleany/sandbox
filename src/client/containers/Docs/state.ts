export interface State {
  markdownContent?: string;
  error?: string;
  isLoading: boolean;
}

export const initialState: State = {
  markdownContent: null,
  error: null,
  isLoading: false,
};

export default State;
