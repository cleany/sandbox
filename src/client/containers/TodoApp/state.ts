import { Todo, Input } from './types';

export interface State {
  todos?: Todo[];
  input?: Input;
}

export const initialState: State = {
  todos: [],
  input: '',
};

export default State;
