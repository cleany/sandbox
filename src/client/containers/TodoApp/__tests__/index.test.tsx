import MockProvider, { getMockStore } from 'redux-mock-provider';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';
import * as React from 'react';
import { initialState } from '../state';
import TodoAppContainer from '../';
import TodoAppPresentation from '../presentation';
import props from './__mocks__/presentation.mock';

const store = getMockStore({
  key: 'todoApp',
  state: initialState,
});

describe('<TodoApp /> container', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <MockProvider store={store}>
        <TodoAppContainer />
      </MockProvider>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});

describe('<TodoAppPresentation />', () => {
  it('should render as expected', () => {
    const wrapper = shallow(
      <TodoAppPresentation {...props} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
