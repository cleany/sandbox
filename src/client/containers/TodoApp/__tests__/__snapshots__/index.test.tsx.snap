// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoApp /> container should render as expected 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(TodoApp) />
</Provider>
`;

exports[`<TodoAppPresentation /> should render as expected 1`] = `
<styled.div>
  <styled.h1>
    Example Todo App
  </styled.h1>
  <styled.div>
    <styled.section>
      <styled.header>
        <styled.input
          autoFocus={true}
          onChange={[Function]}
          onKeyPress={[Function]}
          placeholder="Start typing to add a Todo..."
          value="foo"
        />
      </styled.header>
      <styled.section>
        <styled.ul>
          <styled.li>
            <styled.label>
              foo
            </styled.label>
            <styled.button
              onClick={[Function]}
            />
          </styled.li>
          <styled.li>
            <styled.label>
              bar
            </styled.label>
            <styled.button
              onClick={[Function]}
            />
          </styled.li>
        </styled.ul>
      </styled.section>
    </styled.section>
  </styled.div>
</styled.div>
`;
